File: java/src/main/java/io/github/messageflow/ChainSpan.java
package io.github.messageflow;

/**
 * A unit of tracing work covering the execution of the remainder of a chain.
 */
public interface ChainSpan extends AutoCloseable {

    /** Marks the span as successfully completed. */
    void setOk();

    /**
     * Marks the span as failed.
     *
     * @param throwable the exception that failed the request
     */
    void setError(Throwable throwable);

    /** Ends the span. */
    @Override
    void close();
}

File: node/src/ChainSpan.ts
/**
 * A unit of tracing work covering the execution of the remainder of a chain.
 */
export interface ChainSpan {
  /**
   * Marks the span as successfully completed.
   */
  setOk(): void;

  /**
   * Marks the span as failed.
   *
   * @param error - the error that failed the request
   */
  setError(error: unknown): void;

  /**
   * Ends the span.
   */
  close(): void;
}
